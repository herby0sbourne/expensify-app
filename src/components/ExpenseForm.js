import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';

// const now = moment();
// console.log(now.format('MMM Do, YYYY'));

export default class ExpenseForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      description: props.expense ? props.expense.description : '',
      note: props.expense ? props.expense.note : '',
      amount: props.expense ? (props.expense.amount / 100).toString() : '',
      createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
      calendarFocused: false,
      error: '',
      buttonTitle: props.expense ? 'Edit Expense' : 'Add Expense',
    };
  }

  onDescriptionChange = (e) => {
    const description = e.target.value;

    this.setState(() => {
      return {
        description,
      };
    });
  };

  onNoteChange = (e) => {
    const note = e.target.value;

    this.setState(() => {
      return {
        note,
      };
    });
  };

  onAmountChange = (e) => {
    const amount = e.target.value;

    if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
      // console.log(
      //   new Intl.NumberFormat('en-us', {
      //     style: 'currency',
      //     currency: 'USD',
      //   }).format(amount)
      // );
      this.setState(() => ({ amount }));
    }
  };

  onDateChange = (createdAt) => {
    if (createdAt) {
      this.setState(() => {
        return {
          createdAt,
        };
      });
    }
  };

  onFocusChange = ({ focused }) => {
    this.setState(() => {
      return {
        calendarFocused: focused,
      };
    });
  };

  onSubmit = (e) => {
    e.preventDefault();

    if (!this.state.description || !this.state.amount) {
      //* set error state equal = Please Provide descritption and amount.
      this.setState(() => ({
        error: 'Please Provide descritption and amount.',
      }));
    } else {
      this.setState(() => ({ error: '' }));
      this.props.onSubmit({
        description: this.state.description,
        amount: parseFloat(this.state.amount, 10) * 100,
        createdAt: this.state.createdAt.valueOf(),
        note: this.state.note,
      });
      // console.log('Submitted');
    }
  };

  render() {
    return (
      <form className="form" onSubmit={this.onSubmit}>
        {this.state.error && <p className="form__error">{this.state.error}</p>}
        <input
          className="text-input"
          type="text"
          placeholder="Description"
          autoFocus={true}
          value={this.state.description}
          onChange={this.onDescriptionChange}
        />

        <input
          className="text-input"
          type="text"
          placeholder="Amount"
          value={this.state.amount}
          onChange={this.onAmountChange}
        />

        <SingleDatePicker
          date={this.state.createdAt}
          onDateChange={this.onDateChange}
          focused={this.state.calendarFocused}
          onFocusChange={this.onFocusChange}
          numberOfMonths={1}
          isOutsideRange={() => false}
        />

        <textarea
          className="textarea"
          value={this.state.note}
          placeholder="Add a note for your expense (optional)"
          onChange={this.onNoteChange}
        ></textarea>
        <div>
          <button className="btn">{this.state.buttonTitle}</button>
          {/* <button className="btn">Add Expense</button> */}
        </div>
      </form>
    );
  }
}
