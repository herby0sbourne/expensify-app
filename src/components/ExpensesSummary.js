import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectExpenses from '../selectors/expensesFilter';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
  const expenseWord = expenseCount <= 1 ? 'expense' : 'expenses';
  expensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
  // console.log(expenseCount, expensesTotal);
  return (
    <div className="page-header">
      <div className="container">
        <h1 className="page-header__title">
          Viewing <span>{expenseCount}</span> {expenseWord} total:{' '}
          <span>{expensesTotal}</span>
        </h1>
        <div className="page-header__actions">
          <Link className="btn" to="/create">
            Add Expense
          </Link>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses),
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
