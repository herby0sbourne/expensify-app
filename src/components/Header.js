import React from 'react';
import { connect } from 'react-redux';
import { NavLink, Link } from 'react-router-dom';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => {
  return (
    <header>
      <div className="container">
        <div className="header__content">
          <Link className="header__title" to="/dashboard">
            <h1>Expensify</h1>
          </Link>
          <button className="btn btn-link" onClick={startLogout}>
            Logout
          </button>
        </div>
      </div>
      {/* <NavLink to="/create" activeClassName="is-active">
        Create Expense
      </NavLink> */}

      {/* <NavLink to="/help" activeClassName="is-active">
        Help
      </NavLink> */}
    </header>
  );
};

const mapDispatchToProps = (dispatch) => {
  return {
    startLogout: () => dispatch(startLogout()),
  };
};

export default connect(undefined, mapDispatchToProps)(Header);
