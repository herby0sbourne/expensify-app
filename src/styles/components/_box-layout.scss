.box-layout {
  background: url('/images/bg.jpg') no-repeat center / cover;
  height: 100vh;
  width: 100vw;
  display: flex;
  justify-content: center;
  align-items: center;
}

.box-layout__box {
  background: fade-out(white, 0.5);
  border-radius: 3px;
  text-align: center;
  width: 25rem;
  padding: $l-size $m-size;
}

.box-layout__title {
  margin-bottom: $m-size;
  line-height: 1;
}
