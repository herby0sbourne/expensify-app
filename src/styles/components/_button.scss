button {
  outline: none;
  border: none;
  border-radius: 0.5rem;
}

.btn {
  line-height: 1;
  padding: $s-size;
  font-size: $font-size-large;
  font-weight: 300;
  background: $blue;
  color: white;
  margin-top: $m-size;
  text-decoration: none;
  display: inline-block;
}

.btn.btn-link {
  background: none;
}

.btn.btn-secondary {
  background: #888888;
}
