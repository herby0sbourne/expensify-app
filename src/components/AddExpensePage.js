import React from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { startAddExpense } from '../actions/expenses';

export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.startAddExpense(expense);
    this.props.history.push('/');
  };

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="container">
            <h1 className="page-header__title">Add Expense</h1>
          </div>
        </div>
        <div className="container">
          <ExpenseForm onSubmit={this.onSubmit} />
        </div>
      </div>
    );
  }
}

// const AddExpensePage = (props) => {
//   return (
//     <div>
//       <h1>Add Expense Page</h1>
//       <div>
//         <ExpenseForm
//           onSubmit={(expense) => {
//             // props.dispatch(addExpense(expense));
//             props.onSubmit(expense);
//             props.history.push('/');
//           }}
//         />
//       </div>
//     </div>
//   );
// };

const mapDispatchToProps = (dispatch) => {
  return {
    startAddExpense: (expense) => dispatch(startAddExpense(expense)),
  };
};

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
