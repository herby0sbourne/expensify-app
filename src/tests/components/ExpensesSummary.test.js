import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expensesData from '../fixtures/expensesTestData';

test('should correctly render ExpensesSummary with 1 expense', () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={1} expensesTotal={9434} />
  );
  expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpensesSummary with multiple expenses', () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={2} expensesTotal={9434} />
  );
  expect(wrapper).toMatchSnapshot();
});
