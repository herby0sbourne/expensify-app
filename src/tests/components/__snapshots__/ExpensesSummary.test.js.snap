// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       total:
       
      <span>
        $94.34
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="btn"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should correctly render ExpensesSummary with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        2
      </span>
       
      expenses
       total:
       
      <span>
        $94.34
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="btn"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
